tests/platform_tests/api/test_module.py

```diff
@@ -201,92 +201,92 @@ def test_components(self, duthost, localhost, platform_api_conn):
             pytest.skip("No modules found on device")
 
         # TODO: Ensure the number of components and that the returned list is correct for this platform
-        for i in range(self.num_modules):
+        for mod_idx in range(self.num_modules):
             try:
-                num_components = int(module.get_num_components(platform_api_conn, i))
+                num_components = int(module.get_num_components(platform_api_conn, mod_idx))
             except:
-                pytest.fail("Module {}: num_components is not an integer".format(i))
+                pytest.fail("Module {}: num_components is not an integer".format(mod_idx))
 
-            component_list = chassis.get_all_components(platform_api_conn)
-            pytest_assert(component_list is not None, "Module {}: Failed to retrieve components".format(i))
-            pytest_assert(isinstance(component_list, list) and len(component_list) == num_components, "Module {}: Components appear to be incorrect".format(i))
+            component_list = module.get_all_components(platform_api_conn, mod_idx)
```

Comment:
This seems to be a bug fix other than LGTM?

Reply:
It's a mix of both. This code was copy/pasted from the Chassis class tests, but required a bit of modification. It appears my final commit didn't make it into the original PR, so some calls were inconsistent.